swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift

```diff
@@ -553,6 +632,35 @@ public final class MenuBar: NSObject, ObservableObject {
     }
   }
 
+  @objc private func addFavoriteTapped(_ sender: NSMenuItem) {
+    guard let id = sender.representedObject as? String else { return }
+    setFavorited(id: id, favorited: true)
+  }
+
+  @objc private func removeFavoriteTapped(_ sender: NSMenuItem) {
+    guard let id = sender.representedObject as? String else { return }
+    setFavorited(id: id, favorited: false)
+  }
+
+  private func setFavorited(id: String, favorited: Bool) {
+    if favorited {
+      favorites.add(id)
+    } else {
+      favorites.remove(id)
+    }
```

Comment:
We are already keeping state in `favorites` for a resource being favorited, does it make sense duplicating this in the resource?